cmd/serve/ftp test failure on linux/386

<!--

We understand you are having a problem with rclone; we want to help you with that!

**STOP and READ**
**YOUR POST WILL BE REMOVED IF IT IS LOW QUALITY**:
Please show the effort you've put in to solving the problem and please be specific.
People are volunteering their time to help! Low effort posts are not likely to get good answers!

If you think you might have found a bug, try to replicate it with the latest beta (or stable).
The update instructions are available at https://rclone.org/commands/rclone_selfupdate/

If you can still replicate it or just got a question then please use the rclone forum:

    https://forum.rclone.org/

for a quick response instead of filing an issue on this repo.

If nothing else helps, then please fill in the info below which helps us help you.

**DO NOT REDACT** any information except passwords/keys/personal info.

You should use 3 backticks to begin and end your paste to make it readable.

Make sure to include a log obtained with '-vv'.

You can also use '-vv --log-file bug.log' and a service such as https://pastebin.com or https://gist.github.com/

Thank you

The Rclone Developers

-->

#### What is the problem you are having with rclone?

The test for `cmd/serve/ftp` fails since 29c6c86c009a25c26dde4639766f171164f697f3 (according to `git bisect`).

#### Which OS you are using and how many bits

Void Linux i686, go1.17.2
Alpine 3.14 i386 (container), go1.16.8

#### The command you were trying to run

```
$ git checkout 454574e2cc974a3cacff6ea48e3eeb899d592ec5
$ RCLONE_CONFIG="/notfound" go test -v ./cmd/serve/ftp
=== RUN   TestFTP
=== RUN   TestFTP/Normal
2021/11/04 02:20:17 NOTICE: Local file system at /tmp/rclone208673010: Serving FTP on localhost:51780
2021/11/04 02:20:22 ERROR : rclone-test-sulamok2yivohel3yogodag4/: Dir.Remove not empty
2021/11/04 02:20:28 ERROR : rclone-test-sulamok2yivohel3yogodag4/: Dir.Remove failed to remove directory: remove /tmp/rclone208673010/rclone-test-sulamok2yivohel3yogodag4: directory not empty
2021/11/04 02:20:28 ERROR : rclone-test-sulamok2yivohel3yogodag4/: Dir.Remove failed to remove directory: remove /tmp/rclone208673010/rclone-test-sulamok2yivohel3yogodag4: directory not empty
    servetest.go:97: 
        ----------
        === RUN   TestIntegration
            fstests.go:418: Using remote "ftptest:"
        === RUN   TestIntegration/FsCheckWrap
            fstests.go:455: Not a wrapping Fs
        === RUN   TestIntegration/FsCommand
            fstests.go:483: No commands in this remote
        === RUN   TestIntegration/FsRmdirNotFound
        === RUN   TestIntegration/FsString
        === RUN   TestIntegration/FsName
        === RUN   TestIntegration/FsRoot
        === RUN   TestIntegration/FsRmdirEmpty
        === RUN   TestIntegration/FsMkdir
        === RUN   TestIntegration/FsMkdir/FsMkdirRmdirSubdir
        === RUN   TestIntegration/FsMkdir/FsListEmpty
        === RUN   TestIntegration/FsMkdir/FsListDirEmpty
        === RUN   TestIntegration/FsMkdir/FsListRDirEmpty
            fstests.go:378: FS has no ListR interface
        === RUN   TestIntegration/FsMkdir/FsListDirNotFound
        2021/11/04 02:20:20 ERROR : does not exist: error listing: directory not found
        === RUN   TestIntegration/FsMkdir/FsListRDirNotFound
            fstests.go:378: FS has no ListR interface
        === RUN   TestIntegration/FsMkdir/FsEncoding
        === RUN   TestIntegration/FsMkdir/FsEncoding/control_chars
            fstests.go:670: testing "␀␁␂␃␄␅␆␇␈␉␊␋␌␍␎␏␐␑␒␓␔␕␖␗␘␙␚␛␜␝␞␟␡"
        2021/11/04 02:20:20 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/dot
            fstests.go:670: testing "．"
        2021/11/04 02:20:20 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/dot_dot
            fstests.go:670: testing "．．"
        2021/11/04 02:20:20 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/punctuation
            fstests.go:670: testing "!\"#$%&'()*+,-.／:;<=>?@[\\]^_`{|}~"
        2021/11/04 02:20:20 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/leading_space
            fstests.go:670: testing " leading space"
        2021/11/04 02:20:20 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/leading_tilde
            fstests.go:670: testing "~leading tilde"
        2021/11/04 02:20:20 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/leading_CR
            fstests.go:670: testing "␍leading CR"
        2021/11/04 02:20:20 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/leading_LF
            fstests.go:670: testing "␊leading LF"
        2021/11/04 02:20:20 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/leading_HT
            fstests.go:670: testing "␉leading HT"
        2021/11/04 02:20:20 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/leading_VT
            fstests.go:670: testing "␋leading VT"
        2021/11/04 02:20:20 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/leading_dot
            fstests.go:670: testing ".leading dot"
        2021/11/04 02:20:20 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/trailing_space
            fstests.go:670: testing "trailing space "
        2021/11/04 02:20:20 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/trailing_CR
            fstests.go:670: testing "trailing CR␍"
        2021/11/04 02:20:21 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/trailing_LF
            fstests.go:670: testing "trailing LF␊"
        2021/11/04 02:20:21 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/trailing_HT
            fstests.go:670: testing "trailing HT␉"
        2021/11/04 02:20:21 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/trailing_VT
            fstests.go:670: testing "trailing VT␋"
        2021/11/04 02:20:21 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/trailing_dot
            fstests.go:670: testing "trailing dot."
        2021/11/04 02:20:21 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/invalid_UTF-8
            fstests.go:670: testing "invalid utf-8\xfe"
        2021/11/04 02:20:21 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/URL_encoding
            fstests.go:670: testing "test%46.txt"
        2021/11/04 02:20:21 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsNewObjectNotFound
        === RUN   TestIntegration/FsMkdir/FsPutError
        === RUN   TestIntegration/FsMkdir/FsPutZeroLength
        2021/11/04 02:20:22 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsOpenWriterAt
            fstests.go:743: FS has no OpenWriterAt interface
        === RUN   TestIntegration/FsMkdir/FsChangeNotify
            fstests.go:779: FS has no ChangeNotify interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles
        2021/11/04 02:20:22 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        2021/11/04 02:20:22 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        2021/11/04 02:20:22 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListDirFile2
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListRDirFile2
            fstests.go:378: FS has no ListR interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListR
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListRSubdir
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListDirRoot
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListRDirRoot
            fstests.go:378: FS has no ListR interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListSubdir
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListRSubdir#01
            fstests.go:378: FS has no ListR interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListLevel2
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListRLevel2
            fstests.go:378: FS has no ListR interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListFile1
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsNewObject
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsNewObjectCaseInsensitive
            fstests.go:1032: Not Case Insensitive
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListFile1and2
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsNewObjectDir
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsPurge
            fstests.go:1064: FS has no Purge interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsCopy
            fstests.go:1101: FS has no Copier interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsMove
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsDirMove
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsRmdirFull
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsPrecision
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectString
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectFs
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectRemote
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectHashes
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectModTime
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectMimeType
            fstests.go:1339: MimeType method not supported
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectSetModTime
        2021/11/04 02:20:22 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectSize
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectOpen
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectOpenSeek
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectOpenRange
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectPartialRead
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectUpdate
        2021/11/04 02:20:22 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectStorable
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsIsFile
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsIsFileNotFound
        2021/11/04 02:20:22 ERROR : : error listing: directory not found
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FromRoot
            fstests.go:1499: Opening root remote "ftptest:" path "rclone-test-sulamok2yivohel3yogodag4" from "ftptest:rclone-test-sulamok2yivohel3yogodag4"
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FromRoot/List
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FromRoot/ListEntries
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FromRoot/ListR
            fstests.go:1533: FS has no ListR interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FromRoot/Put
        2021/11/04 02:20:22 ERROR : ftp://localhost:51780: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FromRoot/Put/Remove
        === RUN   TestIntegration/FsMkdir/FsPutFiles/PublicLink
            fstests.go:1592: FS has no PublicLinker interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles/SetTier
            fstests.go:392: FS has no SetTier & GetTier interfaces
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectCheckWrap
            fstests.go:1677: Not a wrapping Fs
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectRemove
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectAbout
            fstests.go:1706: FS does not support About
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsPutStream
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsPutStream/0
        2021/11/04 02:20:22 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsPutStream/100
        2021/11/04 02:20:22 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsPutFiles/Internal
        === RUN   TestIntegration/FsMkdir/FsPutFiles/Internal/UploadTimeout
            ftp_internal_test.go:78: Upload got stuck for 5s !
        === RUN   TestIntegration/FsMkdir/FsPutFiles/Internal/TimePrecision
        === RUN   TestIntegration/FsMkdir/FsPutChunked
            fstests.go:1785: *ftp.Fs does not implement SetUploadChunkSizer
        === RUN   TestIntegration/FsMkdir/FsUploadUnknownSize
        === RUN   TestIntegration/FsMkdir/FsUploadUnknownSize/FsPutUnknownSize
        2021/11/04 02:20:28 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsUploadUnknownSize/FsUpdateUnknownSize
        2021/11/04 02:20:28 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        2021/11/04 02:20:28 ERROR : ftp://localhost:51780/rclone-test-sulamok2yivohel3yogodag4: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsRootCollapse
        2021/11/04 02:20:28 ERROR : : Failed to rmdir: directory not found
        2021/11/04 02:20:28 ERROR : : Failed to rmdir: directory not found
        === RUN   TestIntegration/FsShutdown
        --- FAIL: TestIntegration (7.47s)
            --- SKIP: TestIntegration/FsCheckWrap (0.00s)
            --- SKIP: TestIntegration/FsCommand (0.00s)
            --- PASS: TestIntegration/FsRmdirNotFound (0.00s)
            --- PASS: TestIntegration/FsString (0.00s)
            --- PASS: TestIntegration/FsName (0.00s)
            --- PASS: TestIntegration/FsRoot (0.00s)
            --- PASS: TestIntegration/FsRmdirEmpty (0.00s)
            --- FAIL: TestIntegration/FsMkdir (7.45s)
                --- PASS: TestIntegration/FsMkdir/FsMkdirRmdirSubdir (0.03s)
                --- PASS: TestIntegration/FsMkdir/FsListEmpty (0.00s)
                --- PASS: TestIntegration/FsMkdir/FsListDirEmpty (0.00s)
                --- SKIP: TestIntegration/FsMkdir/FsListRDirEmpty (0.00s)
                --- PASS: TestIntegration/FsMkdir/FsListDirNotFound (0.00s)
                --- SKIP: TestIntegration/FsMkdir/FsListRDirNotFound (0.00s)
                --- PASS: TestIntegration/FsMkdir/FsEncoding (0.50s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/control_chars (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/dot (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/dot_dot (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/punctuation (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/leading_space (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/leading_tilde (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/leading_CR (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/leading_LF (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/leading_HT (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/leading_VT (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/leading_dot (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/trailing_space (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/trailing_CR (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/trailing_LF (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/trailing_HT (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/trailing_VT (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/trailing_dot (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/invalid_UTF-8 (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/URL_encoding (0.02s)
                --- PASS: TestIntegration/FsMkdir/FsNewObjectNotFound (0.00s)
                --- PASS: TestIntegration/FsMkdir/FsPutError (1.03s)
                --- PASS: TestIntegration/FsMkdir/FsPutZeroLength (0.02s)
                --- SKIP: TestIntegration/FsMkdir/FsOpenWriterAt (0.00s)
                --- SKIP: TestIntegration/FsMkdir/FsChangeNotify (0.00s)
                --- FAIL: TestIntegration/FsMkdir/FsPutFiles (5.79s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsListDirFile2 (0.01s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/FsListRDirFile2 (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsListR (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsListRSubdir (0.01s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsListDirRoot (0.01s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/FsListRDirRoot (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsListSubdir (0.00s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/FsListRSubdir#01 (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsListLevel2 (0.00s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/FsListRLevel2 (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsListFile1 (0.01s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsNewObject (0.00s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/FsNewObjectCaseInsensitive (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsListFile1and2 (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsNewObjectDir (0.00s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/FsPurge (0.00s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/FsCopy (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsMove (0.09s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsDirMove (0.06s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsRmdirFull (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsPrecision (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectString (0.01s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectFs (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectRemote (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectHashes (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectModTime (0.00s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/ObjectMimeType (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectSetModTime (0.01s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectSize (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectOpen (0.01s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectOpenSeek (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectOpenRange (0.12s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectPartialRead (0.01s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectUpdate (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectStorable (0.01s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsIsFile (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsIsFileNotFound (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FromRoot (0.11s)
                        --- PASS: TestIntegration/FsMkdir/FsPutFiles/FromRoot/List (0.02s)
                        --- PASS: TestIntegration/FsMkdir/FsPutFiles/FromRoot/ListEntries (0.00s)
                        --- SKIP: TestIntegration/FsMkdir/FsPutFiles/FromRoot/ListR (0.00s)
                        --- PASS: TestIntegration/FsMkdir/FsPutFiles/FromRoot/Put (0.07s)
                            --- PASS: TestIntegration/FsMkdir/FsPutFiles/FromRoot/Put/Remove (0.03s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/PublicLink (0.00s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/SetTier (0.00s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/ObjectCheckWrap (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectRemove (0.03s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/ObjectAbout (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsPutStream (0.05s)
                        --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsPutStream/0 (0.02s)
                        --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsPutStream/100 (0.03s)
                    --- FAIL: TestIntegration/FsMkdir/FsPutFiles/Internal (5.02s)
                        --- FAIL: TestIntegration/FsMkdir/FsPutFiles/Internal/UploadTimeout (5.02s)
                        --- PASS: TestIntegration/FsMkdir/FsPutFiles/Internal/TimePrecision (0.00s)
                --- SKIP: TestIntegration/FsMkdir/FsPutChunked (0.00s)
                --- PASS: TestIntegration/FsMkdir/FsUploadUnknownSize (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsUploadUnknownSize/FsPutUnknownSize (0.01s)
                    --- PASS: TestIntegration/FsMkdir/FsUploadUnknownSize/FsUpdateUnknownSize (0.02s)
                --- PASS: TestIntegration/FsMkdir/FsRootCollapse (0.00s)
            --- PASS: TestIntegration/FsShutdown (0.00s)
        === RUN   TestIntegrationProftpd
            ftp_test.go:23: skipping as -remote is set
        --- SKIP: TestIntegrationProftpd (0.00s)
        === RUN   TestIntegrationPureftpd
            ftp_test.go:34: skipping as -remote is set
        --- SKIP: TestIntegrationPureftpd (0.00s)
        === RUN   TestIntegrationVsftpd
            ftp_test.go:45: skipping as -remote is set
        --- SKIP: TestIntegrationVsftpd (0.00s)
        FAIL
        exit status 1
        FAILgithub.com/rclone/rclone/backend/ftp7.497s
        ----------
    servetest.go:99: 
        Error Trace:servetest.go:99
                    servetest.go:107
        Error:      Received unexpected error:
                    exit status 1
        Test:       TestFTP/Normal
        Messages:   Running ftp integration tests
2021/11/04 02:20:28 NOTICE: Local file system at /tmp/rclone208673010: Stopping FTP on localhost:51780
=== RUN   TestFTP/AuthProxy
2021/11/04 02:20:28 NOTICE: Serving FTP on localhost:51780
2021/11/04 02:20:35 ERROR : rclone-test-nigoyet6mexaxuy0riyajor3/: Dir.Remove not empty
2021/11/04 02:20:40 ERROR : rclone-test-nigoyet6mexaxuy0riyajor3/: Dir.Remove failed to remove directory: remove /tmp/rclone689925364/rclone-test-nigoyet6mexaxuy0riyajor3: directory not empty
2021/11/04 02:20:40 ERROR : rclone-test-nigoyet6mexaxuy0riyajor3/: Dir.Remove failed to remove directory: remove /tmp/rclone689925364/rclone-test-nigoyet6mexaxuy0riyajor3: directory not empty
    servetest.go:97: 
        ----------
        === RUN   TestIntegration
            fstests.go:418: Using remote "ftptest:"
        === RUN   TestIntegration/FsCheckWrap
            fstests.go:455: Not a wrapping Fs
        === RUN   TestIntegration/FsCommand
            fstests.go:483: No commands in this remote
        === RUN   TestIntegration/FsRmdirNotFound
        === RUN   TestIntegration/FsString
        === RUN   TestIntegration/FsName
        === RUN   TestIntegration/FsRoot
        === RUN   TestIntegration/FsRmdirEmpty
        === RUN   TestIntegration/FsMkdir
        === RUN   TestIntegration/FsMkdir/FsMkdirRmdirSubdir
        === RUN   TestIntegration/FsMkdir/FsListEmpty
        === RUN   TestIntegration/FsMkdir/FsListDirEmpty
        === RUN   TestIntegration/FsMkdir/FsListRDirEmpty
            fstests.go:378: FS has no ListR interface
        === RUN   TestIntegration/FsMkdir/FsListDirNotFound
        2021/11/04 02:20:33 ERROR : does not exist: error listing: directory not found
        === RUN   TestIntegration/FsMkdir/FsListRDirNotFound
            fstests.go:378: FS has no ListR interface
        === RUN   TestIntegration/FsMkdir/FsEncoding
        === RUN   TestIntegration/FsMkdir/FsEncoding/control_chars
            fstests.go:670: testing "␀␁␂␃␄␅␆␇␈␉␊␋␌␍␎␏␐␑␒␓␔␕␖␗␘␙␚␛␜␝␞␟␡"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/dot
            fstests.go:670: testing "．"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/dot_dot
            fstests.go:670: testing "．．"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/punctuation
            fstests.go:670: testing "!\"#$%&'()*+,-.／:;<=>?@[\\]^_`{|}~"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/leading_space
            fstests.go:670: testing " leading space"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/leading_tilde
            fstests.go:670: testing "~leading tilde"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/leading_CR
            fstests.go:670: testing "␍leading CR"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/leading_LF
            fstests.go:670: testing "␊leading LF"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/leading_HT
            fstests.go:670: testing "␉leading HT"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/leading_VT
            fstests.go:670: testing "␋leading VT"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/leading_dot
            fstests.go:670: testing ".leading dot"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/trailing_space
            fstests.go:670: testing "trailing space "
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/trailing_CR
            fstests.go:670: testing "trailing CR␍"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/trailing_LF
            fstests.go:670: testing "trailing LF␊"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/trailing_HT
            fstests.go:670: testing "trailing HT␉"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/trailing_VT
            fstests.go:670: testing "trailing VT␋"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/trailing_dot
            fstests.go:670: testing "trailing dot."
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/invalid_UTF-8
            fstests.go:670: testing "invalid utf-8\xfe"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsEncoding/URL_encoding
            fstests.go:670: testing "test%46.txt"
        2021/11/04 02:20:33 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsNewObjectNotFound
        === RUN   TestIntegration/FsMkdir/FsPutError
        === RUN   TestIntegration/FsMkdir/FsPutZeroLength
        2021/11/04 02:20:34 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsOpenWriterAt
            fstests.go:743: FS has no OpenWriterAt interface
        === RUN   TestIntegration/FsMkdir/FsChangeNotify
            fstests.go:779: FS has no ChangeNotify interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles
        2021/11/04 02:20:34 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        2021/11/04 02:20:34 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        2021/11/04 02:20:34 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListDirFile2
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListRDirFile2
            fstests.go:378: FS has no ListR interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListR
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListRSubdir
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListDirRoot
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListRDirRoot
            fstests.go:378: FS has no ListR interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListSubdir
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListRSubdir#01
            fstests.go:378: FS has no ListR interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListLevel2
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListRLevel2
            fstests.go:378: FS has no ListR interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListFile1
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsNewObject
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsNewObjectCaseInsensitive
            fstests.go:1032: Not Case Insensitive
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsListFile1and2
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsNewObjectDir
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsPurge
            fstests.go:1064: FS has no Purge interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsCopy
            fstests.go:1101: FS has no Copier interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsMove
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsDirMove
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsRmdirFull
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsPrecision
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectString
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectFs
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectRemote
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectHashes
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectModTime
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectMimeType
            fstests.go:1339: MimeType method not supported
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectSetModTime
        2021/11/04 02:20:35 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectSize
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectOpen
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectOpenSeek
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectOpenRange
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectPartialRead
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectUpdate
        2021/11/04 02:20:35 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectStorable
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsIsFile
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsIsFileNotFound
        2021/11/04 02:20:35 ERROR : : error listing: directory not found
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FromRoot
            fstests.go:1499: Opening root remote "ftptest:" path "rclone-test-nigoyet6mexaxuy0riyajor3" from "ftptest:rclone-test-nigoyet6mexaxuy0riyajor3"
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FromRoot/List
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FromRoot/ListEntries
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FromRoot/ListR
            fstests.go:1533: FS has no ListR interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FromRoot/Put
        2021/11/04 02:20:35 ERROR : ftp://localhost:51780: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FromRoot/Put/Remove
        === RUN   TestIntegration/FsMkdir/FsPutFiles/PublicLink
            fstests.go:1592: FS has no PublicLinker interface
        === RUN   TestIntegration/FsMkdir/FsPutFiles/SetTier
            fstests.go:392: FS has no SetTier & GetTier interfaces
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectCheckWrap
            fstests.go:1677: Not a wrapping Fs
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectRemove
        === RUN   TestIntegration/FsMkdir/FsPutFiles/ObjectAbout
            fstests.go:1706: FS does not support About
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsPutStream
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsPutStream/0
        2021/11/04 02:20:35 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsPutFiles/FsPutStream/100
        2021/11/04 02:20:35 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsPutFiles/Internal
        === RUN   TestIntegration/FsMkdir/FsPutFiles/Internal/UploadTimeout
            ftp_internal_test.go:78: Upload got stuck for 5s !
        === RUN   TestIntegration/FsMkdir/FsPutFiles/Internal/TimePrecision
        === RUN   TestIntegration/FsMkdir/FsPutChunked
            fstests.go:1785: *ftp.Fs does not implement SetUploadChunkSizer
        === RUN   TestIntegration/FsMkdir/FsUploadUnknownSize
        === RUN   TestIntegration/FsMkdir/FsUploadUnknownSize/FsPutUnknownSize
        2021/11/04 02:20:40 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsUploadUnknownSize/FsUpdateUnknownSize
        2021/11/04 02:20:40 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        2021/11/04 02:20:40 ERROR : ftp://localhost:51780/rclone-test-nigoyet6mexaxuy0riyajor3: SetModTime is not supported
        === RUN   TestIntegration/FsMkdir/FsRootCollapse
        2021/11/04 02:20:40 ERROR : : Failed to rmdir: directory not found
        2021/11/04 02:20:40 ERROR : : Failed to rmdir: directory not found
        === RUN   TestIntegration/FsShutdown
        --- FAIL: TestIntegration (8.65s)
            --- SKIP: TestIntegration/FsCheckWrap (0.00s)
            --- SKIP: TestIntegration/FsCommand (0.00s)
            --- PASS: TestIntegration/FsRmdirNotFound (0.00s)
            --- PASS: TestIntegration/FsString (0.00s)
            --- PASS: TestIntegration/FsName (0.00s)
            --- PASS: TestIntegration/FsRoot (0.00s)
            --- PASS: TestIntegration/FsRmdirEmpty (0.00s)
            --- FAIL: TestIntegration/FsMkdir (7.66s)
                --- PASS: TestIntegration/FsMkdir/FsMkdirRmdirSubdir (0.02s)
                --- PASS: TestIntegration/FsMkdir/FsListEmpty (0.00s)
                --- PASS: TestIntegration/FsMkdir/FsListDirEmpty (0.00s)
                --- SKIP: TestIntegration/FsMkdir/FsListRDirEmpty (0.00s)
                --- PASS: TestIntegration/FsMkdir/FsListDirNotFound (0.00s)
                --- SKIP: TestIntegration/FsMkdir/FsListRDirNotFound (0.00s)
                --- PASS: TestIntegration/FsMkdir/FsEncoding (0.45s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/control_chars (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/dot (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/dot_dot (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/punctuation (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/leading_space (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/leading_tilde (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/leading_CR (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/leading_LF (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/leading_HT (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/leading_VT (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/leading_dot (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/trailing_space (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/trailing_CR (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/trailing_LF (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/trailing_HT (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/trailing_VT (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/trailing_dot (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/invalid_UTF-8 (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsEncoding/URL_encoding (0.03s)
                --- PASS: TestIntegration/FsMkdir/FsNewObjectNotFound (0.00s)
                --- PASS: TestIntegration/FsMkdir/FsPutError (1.04s)
                --- PASS: TestIntegration/FsMkdir/FsPutZeroLength (0.09s)
                --- SKIP: TestIntegration/FsMkdir/FsOpenWriterAt (0.00s)
                --- SKIP: TestIntegration/FsMkdir/FsChangeNotify (0.00s)
                --- FAIL: TestIntegration/FsMkdir/FsPutFiles (5.92s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsListDirFile2 (0.03s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/FsListRDirFile2 (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsListR (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsListRSubdir (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsListDirRoot (0.02s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/FsListRDirRoot (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsListSubdir (0.01s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/FsListRSubdir#01 (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsListLevel2 (0.01s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/FsListRLevel2 (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsListFile1 (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsNewObject (0.00s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/FsNewObjectCaseInsensitive (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsListFile1and2 (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsNewObjectDir (0.00s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/FsPurge (0.00s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/FsCopy (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsMove (0.14s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsDirMove (0.08s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsRmdirFull (0.01s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsPrecision (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectString (0.01s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectFs (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectRemote (0.01s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectHashes (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectModTime (0.00s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/ObjectMimeType (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectSetModTime (0.01s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectSize (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectOpen (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectOpenSeek (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectOpenRange (0.10s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectPartialRead (0.01s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectUpdate (0.03s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectStorable (0.01s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsIsFile (0.01s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsIsFileNotFound (0.01s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FromRoot (0.07s)
                        --- PASS: TestIntegration/FsMkdir/FsPutFiles/FromRoot/List (0.02s)
                        --- PASS: TestIntegration/FsMkdir/FsPutFiles/FromRoot/ListEntries (0.00s)
                        --- SKIP: TestIntegration/FsMkdir/FsPutFiles/FromRoot/ListR (0.00s)
                        --- PASS: TestIntegration/FsMkdir/FsPutFiles/FromRoot/Put (0.05s)
                            --- PASS: TestIntegration/FsMkdir/FsPutFiles/FromRoot/Put/Remove (0.02s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/PublicLink (0.00s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/SetTier (0.00s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/ObjectCheckWrap (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/ObjectRemove (0.03s)
                    --- SKIP: TestIntegration/FsMkdir/FsPutFiles/ObjectAbout (0.00s)
                    --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsPutStream (0.03s)
                        --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsPutStream/0 (0.02s)
                        --- PASS: TestIntegration/FsMkdir/FsPutFiles/FsPutStream/100 (0.02s)
                    --- FAIL: TestIntegration/FsMkdir/FsPutFiles/Internal (5.01s)
                        --- FAIL: TestIntegration/FsMkdir/FsPutFiles/Internal/UploadTimeout (5.01s)
                        --- PASS: TestIntegration/FsMkdir/FsPutFiles/Internal/TimePrecision (0.00s)
                --- SKIP: TestIntegration/FsMkdir/FsPutChunked (0.00s)
                --- PASS: TestIntegration/FsMkdir/FsUploadUnknownSize (0.06s)
                    --- PASS: TestIntegration/FsMkdir/FsUploadUnknownSize/FsPutUnknownSize (0.02s)
                    --- PASS: TestIntegration/FsMkdir/FsUploadUnknownSize/FsUpdateUnknownSize (0.04s)
                --- PASS: TestIntegration/FsMkdir/FsRootCollapse (0.01s)
            --- PASS: TestIntegration/FsShutdown (0.00s)
        === RUN   TestIntegrationProftpd
            ftp_test.go:23: skipping as -remote is set
        --- SKIP: TestIntegrationProftpd (0.00s)
        === RUN   TestIntegrationPureftpd
            ftp_test.go:34: skipping as -remote is set
        --- SKIP: TestIntegrationPureftpd (0.00s)
        === RUN   TestIntegrationVsftpd
            ftp_test.go:45: skipping as -remote is set
        --- SKIP: TestIntegrationVsftpd (0.00s)
        FAIL
        exit status 1
        FAILgithub.com/rclone/rclone/backend/ftp8.668s
        ----------
    servetest.go:99: 
        Error Trace:servetest.go:99
                    servetest.go:110
        Error:      Received unexpected error:
                    exit status 1
        Test:       TestFTP/AuthProxy
        Messages:   Running ftp integration tests
2021/11/04 02:20:40 NOTICE: Stopping FTP on localhost:51780
--- FAIL: TestFTP (23.77s)
    --- FAIL: TestFTP/Normal (11.10s)
    --- FAIL: TestFTP/AuthProxy (12.66s)
FAIL
FAILgithub.com/rclone/rclone/cmd/serve/ftp23.795s
FAIL
```

<!--- Please keep the note below for others who read your bug report. -->

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

--- a/backend/ftp/ftp_internal_test.go
+++ b/backend/ftp/ftp_internal_test.go
@@ -36,7 +36,7 @@ func (f *Fs) testUploadTimeout(t *testing.T) {
 	const (
 		fileSize    = 100000000             // 100 MiB
 		idleTimeout = 40 * time.Millisecond // small because test server is local
-		maxTime     = 5 * time.Second       // prevent test hangup
+		maxTime     = 10 * time.Second      // prevent test hangup
 	)
 
 	if testing.Short() {

